auth-api/src/auth_api/services/affiliation_invitation.py

```diff
@@ -362,10 +363,39 @@ def delete_affiliation_invitation(invitation_id):
         else:
             invitation.delete()
 
+    @staticmethod
+    def _filter_request_invites_role_based(affiliation_invitation_models: List[AffiliationInvitationModel],
+                                           org_id: int,
+                                           user: UserService = None) -> List[AffiliationInvitationModel]:
+        """Filter out affiliation invitations of type REQUEST if current user is not staff or org admin/coordinator."""
+        if UserService.is_context_user_staff():
+            return affiliation_invitation_models
+
+        if org_id is None:
+            return []
+
+        current_user: UserService = user or UserService.find_by_jwt_token()
+        is_org_admin_or_coordinator = UserService.is_user_admin_or_coordinator(user=current_user, org_id=org_id)
+        if is_org_admin_or_coordinator:
+            return affiliation_invitation_models
+
+        # filter out affiliation invitations of type request
+        return list(filter(
+            lambda affiliation_invitation: affiliation_invitation.type != AffiliationInvitationType.REQUEST.value,
+            affiliation_invitation_models))
+
     @staticmethod
     def search_invitations(search_filter: AffiliationInvitationSearch, mask_email=True):
         """Search affiliation invitations."""
-        invitation_models = AffiliationInvitationModel().filter_by(search_filter=search_filter)
+        try:
+            org_id = int(search_filter.to_org_id or search_filter.from_org_id)
```

Comment:
Not sure if this is better or not? 
```
org_id_str = search_filter.to_org_id or search_filter.from_org_id
org_id = int(org_id_str) if org_id_str  and org_id_str .isdecimal() else None
```

Reply:
Yeah, I was trying to decide before writing it. Cause I like how some things are handled for these kind of issues in JS. But I feel like it does not translate very well into the Python. So will leave it as is.